kernel/names.ml

```diff
@@ -34,16 +34,8 @@ struct
 
   let hash = String.hash
 
-  let warn_invalid_identifier =
-    CWarnings.create ~name:"invalid-identifier" ~category:"parsing"
-                     ~default:CWarnings.Disabled
-                     (fun s -> str s)
-
-  let check_soft ?(warn = true) x =
-    let iter (fatal, x) =
-      if fatal then CErrors.error x else
-        if warn then warn_invalid_identifier x
-    in
+  let check_soft ?(strict=true) x =
```

Comment:
Shouldn't this be called `check_identifier`? I thought `soft` meant that we try not to fail when possible, which is not the case here if `strict=true`.